scripts/indexApi.js

```diff
@@ -11,59 +11,59 @@ applist.add = function (app) {
   app.id = Math.random().toString(36).substr(2, 7)
   app.state = 'stopped'
   return applist.findAll()
-  .then(function (apps) {
-    apps.push(app)
-    localStorage.setItem('apps', JSON.stringify(apps))
-    return app
-  })
+    .then(function (apps) {
+      apps.push(app)
+      localStorage.setItem('apps', JSON.stringify(apps))
+      return app
+    })
 }
 applist.find = function (id) {
   return applist.findAll()
-  .then(function (apps) {
-    var findApp = function (app) {
-      return app.id === id
-    }
-    return apps.find(findApp)
-  })
+    .then(function (apps) {
+      var findApp = function (app) {
+        return app.id === id
+      }
+      return apps.find(findApp)
+    })
 }
 
-applist.update = function(app){
+applist.update = function (app) {
   return applist.findAll()
-  .then(function (apps) {
-    var newApps = apps.map(function (currentApp) {
-      if (currentApp.id === app.id) {
-        if(app.name){
-          currentApp.name = app.name
+    .then(function (apps) {
```

Comment:
@gr2m IYou explained this part very well before. Can I ask once more? I'm sorry that I forgot why we need to check 'app id' first and after that we need to check 'app name and state'.(I forgot the logic) I only remember it didn't work before without checking these three elements. The reason was 'app id' is only unique to be able to distinguish the app from other apps? 
```js
 $updateAppForm.on('submit', function (event) {
  event.preventDefault()

  var id = $('#edit-app-container').data('id')
  var newName = $('#rename-app').val()
  var app = {
    id: id,
    name: newName
  }
  if (newName) {
    applist.update(app)
```